exporter/awsxrayexporter/awsxray.go

```diff
@@ -52,27 +52,13 @@ func newTracesExporter(
 		func(ctx context.Context, td pdata.Traces) error {
 			var err error
 			logger.Debug("TracesExporter", typeLog, nameLog, zap.Int("#spans", td.SpanCount()))
-			documents := make([]*string, 0, td.SpanCount())
-			for i := 0; i < td.ResourceSpans().Len(); i++ {
-				rspans := td.ResourceSpans().At(i)
-				resource := rspans.Resource()
-				for j := 0; j < rspans.InstrumentationLibrarySpans().Len(); j++ {
-					spans := rspans.InstrumentationLibrarySpans().At(j).Spans()
-					for k := 0; k < spans.Len(); k++ {
-						document, localErr := translator.MakeSegmentDocumentString(spans.At(k), resource,
-							config.(*Config).IndexedAttributes, config.(*Config).IndexAllAttributes)
-						if localErr != nil {
-							logger.Debug("Error translating span.", zap.Error(localErr))
-							continue
-						}
-						documents = append(documents, &document)
-					}
-				}
-			}
+
+			documents := extractResourceSpans(config, logger, td)
+
 			for offset := 0; offset < len(documents); offset += maxSegmentsPerPut {
 				nextOffset := offset + maxSegmentsPerPut
-				if nextOffset > td.SpanCount() {
-					nextOffset = td.SpanCount()
+				if nextOffset > len(documents) {
+					nextOffset = len(documents)
 				}
```

Comment:
Does it work if we do this?

```go
nextOffset := min(offset + maxSegmentsPerPut, len(documents))
```

`min` may not be the right word... sorry my go knowledge is limited 😅

Reply:
https://stackoverflow.com/questions/27516387/what-is-the-correct-way-to-find-the-min-between-two-integers-in-go go being go does not allow this. min is for float64